accounts-db/src/accounts_cache.rs

```diff
@@ -70,14 +71,14 @@ impl SlotCacheInner {
         &self,
         pubkey: &Pubkey,
         account: AccountSharedData,
-        hash: Option<impl Borrow<Hash>>,
```

Comment:
Can you please, in a pre-pr, remove `hash` from this fn? It is always passed as None from non-test code.

Reply:
I think I'd like to see it before.

Reply:
sorry - more reasoning from my perspective. That `impl Borrow` business came from a desire to not force us to make copies of Vec<Hash> or something at some point in the call chain. It turns out this doesn't matter now (and maybe didn't at that time). But when I look at this fn, it feels like this could require copies of `Vecs` of data on large operations like ancient append vec packing or shrinking. If we make it not possible to go from &Hash to Hash, then we're good. Removing the parameter removes that fear even on changes prior to this.